api/src/main/java/ai/djl/inference/Predictor.java

```diff
@@ -302,40 +308,30 @@ private NDList processInputs(TranslatorContext ctx, List<I> inputs) throws Excep
         return translator.getBatchifier().batchify(preprocessed);
     }
 
-    @SuppressWarnings("PMD.SignatureDeclareThrowsException")
-    private List<O> processOutputs(TranslatorContext ctx, NDList list) throws Exception {
-        NDList[] unbatched = translator.getBatchifier().unbatchify(list);
-        List<O> outputs = new ArrayList<>(unbatched.length);
-        for (NDList output : unbatched) {
-            outputs.add(translator.processOutput(ctx, output));
-        }
-        return outputs;
-    }
-
-    private void preprocessEnd(NDList list) {
+    private void preprocessEnd(NDList list, int batchSize) {
         if (metrics != null) {
             waitToRead(list);
             long tmp = System.nanoTime();
-            long duration = (tmp - timestamp) / 1000;
+            long duration = (tmp - timestamp) / 1000 / batchSize;
```

Comment:
does it make sense to report the average latency per item as the preprocess latency? do we also report the batch size metric somewhere?

I feel like we should report the total time for the full batch, but if this is consistent with how we're doing it elsewhere, or what users typically expect, i'm ok with it. same foes for the other methods we report a similar metric for

Reply:
try to align with TEI for batch case. We do report DynamicBatchSize in metrics. We have total inference time as well